cmake/Modules/FindCBLAS.cmake

```diff
@@ -5,27 +5,54 @@
 #  CBLAS_LIBRARIES   - List of libraries when using CBLAS.
 #  CBLAS_FOUND       - True if CBLAS found.
 
-include (FindPackageHandleStandardArgs)
-
-if (APPLE)
-  find_path (CBLAS_INCLUDES_PRE Accelerate.h)
-  find_library (CBLAS_LIBRARIES_PRE NAMES Accelerate)
-else ()
-  find_path (CBLAS_INCLUDES_PRE cblas.h)
-  find_library (CBLAS_LIBRARIES_PRE NAMES blas)
-endif ()
-
-# handle the QUIETLY and REQUIRED arguments and set CBLAS_FOUND to TRUE if
-# all listed variables are TRUE
-
-if (CBLAS_LIBRARIES_PRE)
-  set (CBLAS_LIBRARIES ${CBLAS_LIBRARIES_PRE})
-endif ()
-if (CBLAS_INCLUDES_PRE)
-  set (CBLAS_INCLUDES ${CBLAS_INCLUDES_PRE})
-  set (CBLAS_FOUND TRUE CACHE INTERNAL "")
-endif ()
+include(FindPackageHandleStandardArgs)
+include(CheckLibraryExists)
 
-find_package_handle_standard_args (CBLAS DEFAULT_MSG CBLAS_LIBRARIES CBLAS_INCLUDES)
+function(FIND_AND_CHECK_CBLAS _libname _includename _symbol)
+  find_path(_CBLAS_INCLUDES ${_includename})
+  find_library(_CBLAS_LIBRARIES NAMES ${_libname})
+
+  # check if cblas symbol is present
+  if(_CBLAS_LIBRARIES AND _CBLAS_INCLUDES)
+    get_filename_component(_LIB_PATH ${_CBLAS_LIBRARIES} DIRECTORY)
+    check_library_exists(${_libname} ${_symbol} ${_LIB_PATH} _HAVE_CBLAS_SYMBOL)
+    if(_HAVE_CBLAS_SYMBOL)
+      set(CBLAS_LIBRARIES ${_CBLAS_LIBRARIES} CACHE FILEPATH "Path to CBLAS library")
+      set(CBLAS_INCLUDES ${_CBLAS_INCLUDES} CACHE PATH "Path to CBLAS include directory")
+    endif(_HAVE_CBLAS_SYMBOL)
+  endif(_CBLAS_LIBRARIES AND _CBLAS_INCLUDES)
+
+  # reset variables
+  unset(_CBLAS_INCLUDES CACHE)
+  unset(_CBLAS_LIBRARIES CACHE)
+endfunction(FIND_AND_CHECK_CBLAS)
+
+
+# check Apple CBLAS
+if(NOT CBLAS_LIBRARIES)
+  FIND_AND_CHECK_CBLAS("Accelerate" "Accelerate/Accelerate.h" cblas_sgemm)
```

Comment:
Apple CBLAS does not work.
You properly need to change `"Accelerate/Accelerate.h"` into `"Accelerate.h"`